src/Workspaces/Core/Portable/Classification/IRemoteSemanticClassificationService.cs

```diff
@@ -82,19 +87,14 @@ private static SerializableClassifiedSpans Dehydrate(ImmutableArray<ClassifiedSp
                 classificationTriples.Add(textSpan.Length);
             }
 
-            return new SerializableClassifiedSpans
-            {
-                ClassificationTypes = classificationTypes,
-                ClassificationTriples = classificationTriples,
-            };
+            return new SerializableClassifiedSpans(
+                classificationTypes.ToImmutableAndClear(),
```

Comment:
>ToImmutableAndClear [](http://example.com/codeflow?start=36&length=19)

Another learning question, why do the "AndClear" part here? The Dispose calls free, which either clears it out if it stays in the pool or removes it from the pool, allowing GC to collect it.

I had originally thought that maybe the Clear would actually null out the values in the array in addition to setting the count to zero, but that doesn't seem to happen, so I'm stumped.

Reply:
the ToImmutableAndClear allows the builder to *move* the intermediary array directly into the final ImmutableArray if the size of hte array matches how filled it is (which is useful in these cases where the size is pregiven).  The ToImmutable version can't do that (since the values stay around in the builder.

Reply:
ahh, nice